kafka.go

```diff
@@ -47,6 +47,19 @@ func NewKafkaProducer(logger *log.Logger, stats *Stats, config *Config) (NozzleP
 
 	producerConfig.ChannelBufferSize = DefaultChannelBufferSize
 
+	switch config.Kafka.Compression {
+	case "gzip":
+		producerConfig.Producer.Compression = sarama.CompressionGZIP
+	case "snappy":
+		producerConfig.Producer.Compression = sarama.CompressionSnappy
+	case "none":
```

Comment:
[nit] Would we like to do `producerConfig.Producer.Compression = sarama.CompressionNone` explicitly in case of default will change in future?

Reply:
unit tests cover this